modules/apps/site/site-admin-web/src/main/resources/META-INF/resources/site_settings/pages.jsp

```diff
@@ -68,177 +68,181 @@ if (!LanguageUtil.isInheritLocales(siteGoup.getGroupId()) && !siteAdminConfigura
-						</c:if>
-
-						<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" helpMessage="enable-propagation-of-changes-from-the-site-template-help" inlineLabel="right" label="enable-propagation-of-changes-from-the-site-template" labelCssClass="simple-toggle-switch" name="privateLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= privateLayoutSetPrototypeLinkEnabled %>" />
-					</div>
+				<c:when test="<%= siteGoup.getPrivateLayoutsPageCount() > 0 %>">
+					<aui:a href="<%= siteGoup.getDisplayURL(themeDisplay, true) %>" label="open-private-pages" target="_blank" />
 				</c:when>
 				<c:otherwise>
-					<aui:input name="privateLayoutSetPrototypeLinkEnabled" type="hidden" value="<%= true %>" />
+					<liferay-ui:message key="this-site-does-not-have-any-private-pages" />
 				</c:otherwise>
 			</c:choose>
-		</c:if>
-	</c:when>
-	<c:otherwise>
-		<c:choose>
-			<c:when test="<%= siteGoup.getPrivateLayoutsPageCount() > 0 %>">
-				<aui:a href="<%= siteGoup.getDisplayURL(themeDisplay, true) %>" label="open-private-pages" target="_blank" />
-			</c:when>
-			<c:otherwise>
-				<liferay-ui:message key="this-site-does-not-have-any-private-pages" />
-			</c:otherwise>
-		</c:choose>
-
-		<c:choose>
-			<c:when test="<%= (privateLayoutSetPrototype != null) && !siteGoup.isStaged() && hasUnlinkLayoutSetPrototypePermission %>">
-				<c:if test="<%= disableLayoutSetPrototypeInput %>">
-					<div class="alert alert-info">
-						<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
-					</div>
-				</c:if>
 
-				<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" inlineLabel="right" label='<%= LanguageUtil.format(request, "enable-propagation-of-changes-from-the-site-template-x", HtmlUtil.escape(privateLayoutSetPrototype.getName(locale)), false) %>' labelCssClass="simple-toggle-switch" name="privateLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= privateLayoutSetPrototypeLinkEnabled %>" />
+			<c:choose>
+				<c:when test="<%= (privateLayoutSetPrototype != null) && !siteGoup.isStaged() && hasUnlinkLayoutSetPrototypePermission %>">
+					<c:if test="<%= disableLayoutSetPrototypeInput %>">
+						<div class="alert alert-info">
+							<liferay-ui:message key="you-cannot-enable-the-propagation-of-changes-because-you-modified-the-display-settings-of-this-site" />
+						</div>
+					</c:if>
 
-				<div class="<%= privateLayoutSetPrototypeLinkEnabled ? "" : "hide" %>" id="<portlet:namespace />privateLayoutSetPrototypeMergeAlert">
+					<aui:input disabled="<%= disableLayoutSetPrototypeInput %>" inlineLabel="right" label='<%= LanguageUtil.format(request, "enable-propagation-of-changes-from-the-site-template-x", HtmlUtil.escape(privateLayoutSetPrototype.getName(locale)), false) %>' labelCssClass="simple-toggle-switch" name="privateLayoutSetPrototypeLinkEnabled" type="toggle-switch" value="<%= privateLayoutSetPrototypeLinkEnabled %>" />
```

Comment:
This line adds a form input. Please check:

* The input is validated on the processing action

* The backend is protected agains value forging



Refer to documentation: